allen-cell-animated/simularium-viewer
# Fix/warn if waiting too long to connect to server

Problem
=======
Closes #169 

Solution
========
The timeout for connecting to the server was already built into `RemoteSimulator` but the error message was not getting propagated up as a `FrontEndError`. So I made that error bubble up from `connectToRemoteServer` to `startRemoteTrajectoryPlayback` where it would be thrown as a `FrontEndError` that would be displayed in the UI. I also added some tests.

While working on the above I refactored `connectToRemoteServer` to be more testable and hopefully a little clearer in general.

Thanks to @meganrm for help writing tests!

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)
* This change requires updated or new tests

Steps to Verify:
----------------
1. Test should pass
2. `npm start` and check that networked models load in the viewer testbed with no new problems

Keyfiles (delete if not relevant):
-----------------------
1. src/simularium/RemoteSimulator.ts
2. src/test/RemoteSimulator.test.ts

Thanks for contributing!
